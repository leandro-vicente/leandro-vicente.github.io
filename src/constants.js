const SOCIAL_ICONS_AND_LINKS = [
  {
    'url': 'https://www.linkedin.com/in/leandrovicentesilva/',
    'name': 'linkedin'
  },
  {
    'glyphicon': 'github',
    'url': 'https://github.com/llevvi/',
    'name': 'github'
  },
  {
    'url': 'http://youtube.com/levvitalk',
    'name': 'youtube'
  },
  {
    'url': 'http://instagram.com/levvitalk',
    'name': 'instagram'
  }
]

const SUBSCRIBE_FORM_URL = 'https://goo.gl/forms/A3zRVgzeH3BMD2FJ3'
  
export { SOCIAL_ICONS_AND_LINKS, SUBSCRIBE_FORM_URL }