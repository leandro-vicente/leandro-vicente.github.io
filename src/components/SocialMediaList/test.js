import React from 'react'
import SocialMediaList from './index'
import { SOCIAL_ICONS_AND_LINKS } from '../../constants';
import { shallow } from 'enzyme'

describe('SocialMediaList', () => {
  it('renders as expected', () => {
    const socialMedia = shallow(<SocialMediaList iconsAndLinks={SOCIAL_ICONS_AND_LINKS} />)
    expect(socialMedia).toMatchSnapshot()
  })
})