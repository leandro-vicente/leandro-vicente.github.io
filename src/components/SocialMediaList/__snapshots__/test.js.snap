// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaList renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialMediaList
    iconsAndLinks={
        Array [
            Object {
              "glyphicon": "fa fa-linkedin fa-2x",
              "name": "linkedin",
              "url": "https://www.linkedin.com/in/leandrovicentesilva/",
            },
            Object {
              "glyphicon": "fa fa-github fa-2x",
              "name": "github",
              "url": "https://github.com/llevvi/",
            },
            Object {
              "glyphicon": "fa fa-youtube-play fa-2x",
              "name": "youtube",
              "url": "http://youtube.com/levvitalk",
            },
            Object {
              "glyphicon": "fa fa-instagram fa-2x",
              "name": "instagram",
              "url": "http://instagram.com/levvitalk",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <a
                    aria-label="linkedin link"
                    className="fa fa-linkedin fa-2x"
                    href="https://www.linkedin.com/in/leandrovicentesilva/"
                    target="_blank"
          />
</li>,
        <li>
          <a
                    aria-label="github link"
                    className="fa fa-github fa-2x"
                    href="https://github.com/llevvi/"
                    target="_blank"
          />
</li>,
        <li>
          <a
                    aria-label="youtube link"
                    className="fa fa-youtube-play fa-2x"
                    href="http://youtube.com/levvitalk"
                    target="_blank"
          />
</li>,
        <li>
          <a
                    aria-label="instagram link"
                    className="fa fa-instagram fa-2x"
                    href="http://instagram.com/levvitalk"
                    target="_blank"
          />
</li>,
      ],
      "className": "social-media-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <a
            aria-label="linkedin link"
            className="fa fa-linkedin fa-2x"
            href="https://www.linkedin.com/in/leandrovicentesilva/"
            target="_blank"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "linkedin link",
            "className": "fa fa-linkedin fa-2x",
            "href": "https://www.linkedin.com/in/leandrovicentesilva/",
            "target": "_blank",
          },
          "ref": null,
          "rendered": null,
          "type": "a",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <a
            aria-label="github link"
            className="fa fa-github fa-2x"
            href="https://github.com/llevvi/"
            target="_blank"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "github link",
            "className": "fa fa-github fa-2x",
            "href": "https://github.com/llevvi/",
            "target": "_blank",
          },
          "ref": null,
          "rendered": null,
          "type": "a",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <a
            aria-label="youtube link"
            className="fa fa-youtube-play fa-2x"
            href="http://youtube.com/levvitalk"
            target="_blank"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "youtube link",
            "className": "fa fa-youtube-play fa-2x",
            "href": "http://youtube.com/levvitalk",
            "target": "_blank",
          },
          "ref": null,
          "rendered": null,
          "type": "a",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": <a
            aria-label="instagram link"
            className="fa fa-instagram fa-2x"
            href="http://instagram.com/levvitalk"
            target="_blank"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "instagram link",
            "className": "fa fa-instagram fa-2x",
            "href": "http://instagram.com/levvitalk",
            "target": "_blank",
          },
          "ref": null,
          "rendered": null,
          "type": "a",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <a
                        aria-label="linkedin link"
                        className="fa fa-linkedin fa-2x"
                        href="https://www.linkedin.com/in/leandrovicentesilva/"
                        target="_blank"
            />
</li>,
          <li>
            <a
                        aria-label="github link"
                        className="fa fa-github fa-2x"
                        href="https://github.com/llevvi/"
                        target="_blank"
            />
</li>,
          <li>
            <a
                        aria-label="youtube link"
                        className="fa fa-youtube-play fa-2x"
                        href="http://youtube.com/levvitalk"
                        target="_blank"
            />
</li>,
          <li>
            <a
                        aria-label="instagram link"
                        className="fa fa-instagram fa-2x"
                        href="http://instagram.com/levvitalk"
                        target="_blank"
            />
</li>,
        ],
        "className": "social-media-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <a
              aria-label="linkedin link"
              className="fa fa-linkedin fa-2x"
              href="https://www.linkedin.com/in/leandrovicentesilva/"
              target="_blank"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "linkedin link",
              "className": "fa fa-linkedin fa-2x",
              "href": "https://www.linkedin.com/in/leandrovicentesilva/",
              "target": "_blank",
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <a
              aria-label="github link"
              className="fa fa-github fa-2x"
              href="https://github.com/llevvi/"
              target="_blank"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "github link",
              "className": "fa fa-github fa-2x",
              "href": "https://github.com/llevvi/",
              "target": "_blank",
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <a
              aria-label="youtube link"
              className="fa fa-youtube-play fa-2x"
              href="http://youtube.com/levvitalk"
              target="_blank"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "youtube link",
              "className": "fa fa-youtube-play fa-2x",
              "href": "http://youtube.com/levvitalk",
              "target": "_blank",
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": <a
              aria-label="instagram link"
              className="fa fa-instagram fa-2x"
              href="http://instagram.com/levvitalk"
              target="_blank"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "instagram link",
              "className": "fa fa-instagram fa-2x",
              "href": "http://instagram.com/levvitalk",
              "target": "_blank",
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
